modules/nickserv/identify.c

```diff
@@ -90,11 +92,17 @@ static void ns_cmd_login(sourceinfo_t *si, int parc, char *parv[])
 		return;
 	}
 
-	req.si = si;
-	req.mu = mu;
-	req.allowed = true;
-	hook_call_user_can_login(&req);
-	if (!req.allowed)
+	login_req.si = si;
+	login_req.mu = mu;
+	login_req.allowed = true;
+	logout_req.si = si;
+	logout_req.u = u;
+	logout_req.allowed = true;
+	hook_call_user_can_login(&login_req);
+	if (login_req.allowed && u->myuser)
+		hook_call_user_can_logout(&logout_req);
+
+	if (!login_req.allowed || !logout_req.allowed)
```

Comment:
Wouldn't the text beneath be wrong here? If I am not permitted to log out services would send me "You cannt log in as %s because ..."

Reply:
This is about logging in via `/ns identify` here. If you are already identified, it checks whether you are permitted from logging out of the old account and then logging in again. So technically the text is correct.

The actual question would be though, if this should be classified as "logging out, then logging in" or actually "change account (name)".

Reply:
I wonder if it's worth providing the `user_can_logout` hook with an additional parameter specifying whether the "logout" will actually leave the user not logged in to any account (as opposed to merely changing accounts).